js/src/duplicate-post-elementor.js

```diff
@@ -0,0 +1,89 @@
+/* global $e */
+
+/**
+ * Hooks into Elementor on initialization.
+ *
+ * @returns {void}
+ */
+function duplicatePostOnElementorInitialize() {
+
+    /**
+     * Class that defines the redirect action to execute after republishing.
+     */
+    class RedirectAfterRepublish extends $e.modules.hookUI.After {
+
+        /**
+         * Gets the command to run on.
+         *
+         * @returns {string} The command.
+         */
+        getCommand() {
+            return "document/save/save";
+        }
+
+        /**
+         * Gets the conditions on which to run on.
+         *
+         * @param {Object} args The arguments to use.
+         *
+         * @returns {boolean} Whether the post status is that of a published post.
+         */
+        getConditions( args ) {
+            const { status } = args;
+
+            return status === "publish";
+        }
+
+        /**
+         * Gets the ID of the action.
+         *
+         * @returns {string} The ID.
+         */
+        getId() {
+            return "redirect-after-republish";
+        }
+
+        /**
+         * Applies the redirect if the condtions are met.
+         *
+         * @param {Object} args The arguments to use.
+         *
+         * @returns {void}
+         */
+        apply( args ) {
+            if ( args.status === "publish" && duplicatePost.originalEditURL ) {
```

Comment:
`duplicatePost` is a global, it should be added in the comment at the top of the file.